sycl/include/sycl/detail/string_view.hpp

```diff
@@ -0,0 +1,66 @@
+//==----------------- string_view.hpp - SYCL standard header file
+//---------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+namespace sycl {
+inline namespace _V1 {
+namespace detail {
+
+// This class  and detail::string are intended to support different ABIs
+// between libsycl and the user program. This class is not inteded to replace
+// std::string for general purpose usage.
+// One most important issue is different ABIs for std::string before
+// C++11-ABI and after C++11-ABI.
+// To address this issue, we define sycl::detail::string_view class.
+// There are two occasions that std::string crosses the ABI boundaries.
+// Once from the user program to the libsycl side, where we need to
+// pass the existing std::string to libsycl. This one uses the member 'str'.
+// Another occasion is returning std::string from libsycl to the user program.
+// In this case, sycl::detail::string provides a placeholder pointer, str.
```

Comment:
because it is important to understand difference between two classes.
detail::string_view is intended to pass user's std::string to libsycl, and 
detail::string is intended to return std::string from the libsycl to the user program.
It should be explained together to understand the restrictions of these two files.